internal/objects/service/service_test.go

```diff
@@ -175,21 +187,54 @@ func TestDesiredEnvoyService(t *testing.T) {
 	// Check AWS NLB load balancer type.
 	awsParams := operatorv1alpha1.ProviderLoadBalancerParameters{
 		Type: operatorv1alpha1.AWSLoadBalancerProvider,
-		AWS:  &operatorv1alpha1.AWSLoadBalancerParameters{Type: operatorv1alpha1.AWSNetworkLoadBalancer},
+		AWS:  &operatorv1alpha1.AWSLoadBalancerParameters{Type: operatorv1alpha1.AWSNetworkLoadBalancer, AllocationIDs: &allocationIDs},
 	}
 	cntr.Spec.NetworkPublishing.Envoy.LoadBalancer.ProviderParameters = awsParams
 	svc = DesiredEnvoyService(cntr)
 	checkServiceHasAnnotation(t, svc, true, awsLBTypeAnnotation)
+	checkServiceHasAnnotation(t, svc, true, awsLBAllocationIDsAnnotation)
 	checkServiceHasAnnotation(t, svc, false, awsLbBackendProtoAnnotation)
+	// Check Azure external load balancer type.
+	azureParams := operatorv1alpha1.ProviderLoadBalancerParameters{
+		Type:  operatorv1alpha1.AzureLoadBalancerProvider,
+		Azure: &operatorv1alpha1.AzureLoadBalancerParameters{Address: &loadBalancerAddress, ResourceGroup: &resourceGroup},
+	}
+	cntr.Spec.NetworkPublishing.Envoy.LoadBalancer.ProviderParameters = azureParams
+	svc = DesiredEnvoyService(cntr)
+	checkServiceHasLoadBalancerAddress(t, svc, loadBalancerAddress)
+	checkServiceHasAnnotation(t, svc, true, azureLBResourceGroupAnnotation)
+	// Check GCP external load balancer type.
+	gcpParams := operatorv1alpha1.ProviderLoadBalancerParameters{
+		Type: operatorv1alpha1.GCPLoadBalancerProvider,
+		GCP:  &operatorv1alpha1.GCPLoadBalancerParameters{Address: &loadBalancerAddress},
+	}
+	cntr.Spec.NetworkPublishing.Envoy.LoadBalancer.ProviderParameters = gcpParams
+	svc = DesiredEnvoyService(cntr)
+	checkServiceHasLoadBalancerAddress(t, svc, loadBalancerAddress)
 	cntr.Spec.NetworkPublishing.Envoy.LoadBalancer.Scope = operatorv1alpha1.InternalLoadBalancer
+	cntr.Spec.NetworkPublishing.Envoy.LoadBalancer.ProviderParameters.Type = operatorv1alpha1.AWSLoadBalancerProvider
 	svc = DesiredEnvoyService(cntr)
 	checkServiceHasAnnotation(t, svc, true, awsInternalLBAnnotation)
-	cntr.Spec.NetworkPublishing.Envoy.LoadBalancer.ProviderParameters.Type = operatorv1alpha1.AzureLoadBalancerProvider
+	// Check Azure internal load balancer type.
+	azureParamsInternal := operatorv1alpha1.ProviderLoadBalancerParameters{
+		Type:  operatorv1alpha1.AzureLoadBalancerProvider,
+		Azure: &operatorv1alpha1.AzureLoadBalancerParameters{Address: &loadBalancerAddress, Subnet: &subnet},
+	}
+	cntr.Spec.NetworkPublishing.Envoy.LoadBalancer.ProviderParameters = azureParamsInternal
 	svc = DesiredEnvoyService(cntr)
+	checkServiceHasLoadBalancerAddress(t, svc, loadBalancerAddress)
 	checkServiceHasAnnotation(t, svc, true, azureInternalLBAnnotation)
-	cntr.Spec.NetworkPublishing.Envoy.LoadBalancer.ProviderParameters.Type = operatorv1alpha1.GCPLoadBalancerProvider
+	checkServiceHasAnnotation(t, svc, true, azureLBSubnetAnnotation)
```

Comment:
`checkServiceHasAnnotation` is now `checkServiceHasAnnotations` and now accepts multiple annotations, so these lines should be merged after rebasing.